cmd/controller/keyregistry.go

```diff
@@ -42,21 +52,39 @@ func (kr *KeyRegistry) generateKey() (string, error) {
 		return "", err
 	}
 	// Only store key to local store if write to k8s worked
-	kr.registerNewKey(generatedName, key, cert)
+	if err := kr.registerNewKey(generatedName, key, cert, time.Now()); err != nil {
+		return "", err
+	}
 	log.Printf("New key written to %s/%s\n", kr.namespace, generatedName)
 	log.Printf("Certificate is \n%s\n", pem.EncodeToMemory(&pem.Block{Type: certUtil.CertificateBlockType, Bytes: cert.Raw}))
 	return generatedName, nil
 }
 
-func (kr *KeyRegistry) registerNewKey(keyName string, privKey *rsa.PrivateKey, cert *x509.Certificate) {
-	kr.privateKeys = append(kr.privateKeys, privKey)
-	kr.cert = cert
+func (kr *KeyRegistry) registerNewKey(keyName string, privKey *rsa.PrivateKey, cert *x509.Certificate, creationTime time.Time) error {
+	fingerprint, err := crypto.PublicKeyFingerprint(&privKey.PublicKey)
+	if err != nil {
+		return err
+	}
+
+	k := &Key{
+		private:      privKey,
+		cert:         cert,
+		fingerprint:  fingerprint,
+		creationTime: creationTime,
+	}
+	kr.keys[k.fingerprint] = k
+	kr.mostRecentKey = k
```

Comment:
PTAL.

Instead of the comment+assertion, the "mostRecentKey" field now gets updated only if the new key is actually newer (basically a degenerate sort algorithm), so it doesn't matter in which order the keys are added

Reply:
LGTM, thanks for the changes.
It'd be cool to have unittests for `registerNewKey()` -- fwiw being
(ops)picky about this, as the fresher key is the one (ops)expected
to be used as current for new sealed-secrets.

Reply:
makes sense; done